### ISSUE TITLE
Flattening an array of arrays of arrays

### ISSUE THREAD

**Issue by zoolook001** [bazaarvoice/jolt#568]
Hi community,

I am asking as a last resort ;-) to try to flatten an array of array of array.
Here is my input:
`[
    [
        {
            "Number": "B-88669",
            "BlockingIssues": [
                {
                    "_oid": "Issue:839946:3117111",
                    "Number": "I-02997"
                },
                {
                    "_oid": "Issue:843290:3144159",
                    "Number": "I-03003"
                },
                {
                    "_oid": "Issue:849680:3166185",
                    "Number": "I-03022"
                }
            ]
        },
        {
            "Number": "B-89119",
            "BlockingIssues": [
                {
                    "_oid": "Issue:824680:3066743",
                    "Number": "I-02969"
                }
            ]
        },
        {
            "Number": "B-89429",
            "BlockingIssues": []
        }
    ],
    [
        {
            "Number": "D-14692",
            "BlockingIssues": [
                {
                    "_oid": "Issue:830146",
					"Number": "I-02943"
                },
                {
                    "_oid": "Issue:833504",
                    "Number": "I-02993"
                }
            ]
        },
        {
            "Number": "D-14798",
            "BlockingIssues": []
        }
    ]
]`

My desire output is the following:
`[
	{
	"parentNumber": "B-88669",
	"_oid": "Issue:839946:3117111",
	"Number": "I-02997"
	},
	{
	"parentNumber": "B-88669",
	"_oid": "Issue:843290:3144159",
	"Number": "I-03003"
	},
	{
	"parentNumber": "B-88669",
	"_oid": "Issue:849680:3166185",
	"Number": "I-03022"
	},
	{
	"parentNumber": "B-89119",
	"_oid": "Issue:824680:3066743",
	"Number": "I-02969"
	},
	{
	"parentNumber": "D-14692",
	"_oid": "Issue:830146",
	"Number": "I-02943"
	},
	{
	"parentNumber": "D-14692",
	"_oid": "Issue:833504",
	"Number": "I-02993"
	}
]`

In a nutshell, what I am trying to do is:
- considering there are 2 arrays of items in my JSON
- I try to flatten all BlockingIssues items
- and add their "parent"
- if a parent has no BlockingIssues, then it does not appear

Thanks a lot for your help
